python_ta/typecheck/base.py

```diff
@@ -505,7 +505,10 @@ def resolve(self, t: type) -> TypeResult:
             try:
                 repr = self.find_repr(self.type_to_tnode[str(t)])
                 if repr and repr.type is not t:
-                    return self.resolve(repr.type)
+                    if getattr(repr.type, '__args__', None) is not None and any(elt is t for elt in repr.type.__args__):
```

Comment:
This can be simplified by using `getattr` directly in the comprehension in the any: `any(elt is t for elt in getattr(...))`, of course choosing a good default value.